src/components/person.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { MyContext } from './state';
+import { Person as PersonType } from '../types';
+
+interface PersonComponentProps {
+  pers: PersonType;
+}
+
+export const Person: React.FC<PersonComponentProps> = ({ pers }) => {
+  const { name, sex, born, fatherName, motherName, slug, died } = pers;
+  const { setSelectedPerson, person } = useContext(MyContext);
+  const location = useLocation();
+
+  const motherPerson = person.find(
+    (perso: PersonType) => perso.name === motherName,
+  );
+  const fatherPerson = person.find(
+    (perso: PersonType) => perso.name === fatherName,
+  );
```

Comment:
The current implementation of finding `motherPerson` and `fatherPerson` is not optimal because it uses `Array.prototype.find` inside the render method, which will run every time the component re-renders, potentially affecting performance. Consider memoizing these values using `useMemo` hook to ensure that these computations are only recalculated when `person`, `motherName`, or `fatherName` change.